pkg/compact/retention_test.go

```diff
@@ -36,17 +43,23 @@ func TestTimeBasedRetentionPolicyOnEmptyBucket(t *testing.T) {
 	testutil.Equals(t, got, want)
 }
 
-func TestTimeBasedRetentionPolicyKeepsBucketsBeforeDuration(t *testing.T) {
+func TestTimeBasedRetentionPolicyByResolutionKeepsBucketsBeforeDuration(t *testing.T) {
 	logger := log.NewNopLogger()
 	bkt := inmem.NewBucket()
 
-	uploadMockBlock(t, bkt, "01CPHBEX20729MJQZXE3W0BW48", time.Now().Add(-3*24*time.Hour), time.Now().Add(-2*24*time.Hour))
-	uploadMockBlock(t, bkt, "01CPHBEX20729MJQZXE3W0BW49", time.Now().Add(-2*24*time.Hour), time.Now().Add(-24*time.Hour))
-	uploadMockBlock(t, bkt, "01CPHBEX20729MJQZXE3W0BW50", time.Now().Add(-24*time.Hour), time.Now().Add(-23*time.Hour))
-	uploadMockBlock(t, bkt, "01CPHBEX20729MJQZXE3W0BW51", time.Now().Add(-23*time.Hour), time.Now().Add(-6*time.Hour))
-	testutil.Ok(t, compact.ApplyDefaultRetentionPolicy(context.TODO(), logger, bkt, 24*time.Hour))
+	retentionsByResolution := map[compact.ResolutionLevel]time.Duration{
+		compact.ResolutionLevelRaw: 24 * time.Hour,
+		compact.ResolutionLevel5m:  0,
+		compact.ResolutionLevel1h:  0,
+	}
 
-	want := []string{"01CPHBEX20729MJQZXE3W0BW50/", "01CPHBEX20729MJQZXE3W0BW51/"}
+	uploadMockBlock(t, bkt, "01CPHBEX20729MJQZXE3W0BW48", time.Now().Add(-3*24*time.Hour), time.Now().Add(-2*24*time.Hour), downsample.ResLevel0)
```

Comment:
Can we upload one that is Res0 and have newer time, close to the edge to see if we are not killling good ones?